gauntlet/packages/gauntlet-solana/src/commands/wallet.ts

```diff
@@ -0,0 +1,96 @@
+const { Wallet } = require('@project-serum/anchor')
+import { Wallet as SerumWallet } from '@project-serum/anchor'
+import { Keypair, PublicKey, Transaction } from '@solana/web3.js'
+import TransportNodeHid from '@ledgerhq/hw-transport-node-hid'
+import Solana from '@ledgerhq/hw-app-solana'
+import { logger } from '@chainlink/gauntlet-core/dist/utils'
+
+export enum WalletTypes {
+  LOCAL = 'local',
+  LEDGER = 'ledger',
+}
+
+export abstract class SolanaWallet {
+  static create: (...args) => Promise<SolanaWallet>
+
+  abstract signTransaction: (tx: Transaction) => Promise<Transaction>
+  abstract signAllTransactions: (txs: Transaction[]) => Promise<Transaction[]>
+  abstract publicKey: PublicKey
+  abstract payer: Keypair
+
+  abstract type: () => WalletTypes
+}
+
+export class LocalWallet extends SolanaWallet {
+  wallet: typeof SerumWallet
+
+  constructor(wallet: typeof SerumWallet) {
+    super()
+    this.wallet = wallet
+  }
+
+  static create = async (keypair: Keypair) => {
+    const wallet = new Wallet(keypair)
+    return new LocalWallet(wallet)
+  }
+
+  signTransaction = (tx: Transaction) => this.wallet.signTransaction(tx)
+  signAllTransactions = (txs: Transaction[]) => this.wallet.signAllTransactions(txs)
+
+  get publicKey() {
+    return this.wallet.payer.publicKey
+  }
+
+  get payer() {
+    return this.wallet.payer
+  }
+
+  type = () => WalletTypes.LOCAL
+}
+
+export class LedgerWallet extends SolanaWallet {
+  publicKey: PublicKey
+  wallet: Solana
+  path: string
+
+  constructor(solanaLW: Solana, pubKey: PublicKey, path: string) {
```

Comment:
By manually allowing to specify both `path` and `pubKey` we introduced a way to misconfigure the class and break invariant.

Let's avoid specifying the `pubKey` for `LedgerWallet` and get it every time directly from Ledger.